src/System.Net.Sockets/tests/PerformanceTests/Perf.Sockets.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Net.Sockets.Tests;
+using System.Net.Test.Common;
+using System.Threading.Tasks;
+
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.Xunit.Performance;
+
+namespace System.Net.Sockets.Performance.Tests
+{
+    public class Perf_Sockets
+    {
+        /// <summary>
+        /// This test is built on top of the existing perf infrastructure and thus has some oddities that 
+        /// may be removed after some cleanup. I did not want to affect the structure of the SocketCommon any more than 
+        /// I had to, so I built this test around the existing API.
+        /// </summary>
+        [Benchmark]
+        [InlineData(0, 256, 100, SocketImplementationType.Async, SocketImplementationType.Async)]
+        [InlineData(0, 256, 1, SocketImplementationType.Async, SocketImplementationType.Async)]
+        [PlatformSpecific(PlatformID.Windows)] // Only tested on Windows&Linux and Linux fails with missing dll "ws2_32.dll"
```

Comment:
Looks like the version I was using wasn't updated recently enough. Using all locally built libraries, the tests pass on Linux and give me the following results:

```
  <test name="System.Net.Sockets.Performance.Tests.Perf_Sockets.SocketTest(port: 0, bufferSize: 256, socketInstances: 100, serverType: Async, clientType: Async)" type="System.Net.Sockets.Performance.Tests.Perf_Sockets" method="SocketTest" time="1.1753764" result="Pass">
	<traits>
	  <trait name="Benchmark" value="true" />
	</traits>
	<performance runid="System.Net.Sockets.Async.Performance.Tests.dll-Linux" etl="/home/ianha/sockets/results/System.Net.Sockets.Async.Performance.Tests.dll-Linux.csv">
	  <metrics>
		<Duration displayName="Duration" unit="msec" />
	  </metrics>
	  <iterations>
		<iteration index="0" Duration="89.010599970817566" />
		<iteration index="1" Duration="15.383499979972839" />
		<iteration index="2" Duration="9.4274001121521" />
		<iteration index="3" Duration="6.3216999769210815" />
		<iteration index="4" Duration="5.5986999273300171" />
		<iteration index="5" Duration="6.0784000158309937" />
		<iteration index="6" Duration="5.8564000129699707" />
		<iteration index="7" Duration="16.754200100898743" />
		<iteration index="8" Duration="7.386199951171875" />
		<iteration index="9" Duration="6.3339999914169312" />
		<iteration index="10" Duration="6.162600040435791" />
		<iteration index="11" Duration="18.2992000579834" />
		<iteration index="12" Duration="5.7546000480651855" />
		<iteration index="13" Duration="5.8536999225616455" />
		<iteration index="14" Duration="5.7855000495910645" />
		<iteration index="15" Duration="5.666100025177002" />
		<iteration index="16" Duration="5.5970999002456665" />
		<iteration index="17" Duration="5.5190999507904053" />
		<iteration index="18" Duration="5.6934000253677368" />
		<iteration index="19" Duration="19.406499981880188" />
		<iteration index="20" Duration="6.4606000185012817" />
		<iteration index="21" Duration="5.6282999515533447" />
		<iteration index="22" Duration="8.1333000659942627" />
		<iteration index="23" Duration="5.6297999620437622" />
		<iteration index="24" Duration="6.3799999952316284" />
		<iteration index="25" Duration="5.7937999963760376" />
		<iteration index="26" Duration="5.59089994430542" />
		<iteration index="27" Duration="5.9428000450134277" />
		<iteration index="28" Duration="6.7113000154495239" />
		<iteration index="29" Duration="6.1858000755310059" />
		<iteration index="30" Duration="6.0367000102996826" />
		<iteration index="31" Duration="6.030500054359436" />
		<iteration index="32" Duration="6.0856000185012817" />
		<iteration index="33" Duration="5.9524999856948853" />
		<iteration index="34" Duration="5.6888999938964844" />
		<iteration index="35" Duration="16.741100072860718" />
		<iteration index="36" Duration="5.615399956703186" />
		<iteration index="37" Duration="5.8831000328063965" />
		<iteration index="38" Duration="6.1351999044418335" />
		<iteration index="39" Duration="5.6634000539779663" />
		<iteration index="40" Duration="6.1261999607086182" />
		<iteration index="41" Duration="5.7696999311447144" />
		<iteration index="42" Duration="5.8904000520706177" />
		<iteration index="43" Duration="5.8809000253677368" />
		<iteration index="44" Duration="6.0446000099182129" />
		<iteration index="45" Duration="5.7204000949859619" />
		<iteration index="46" Duration="5.7815999984741211" />
		<iteration index="47" Duration="5.9017000198364258" />
		<iteration index="48" Duration="5.9829999208450317" />
		<iteration index="49" Duration="5.773900032043457" />
		<iteration index="50" Duration="5.8238999843597412" />
	  </iterations>
	</performance>
  </test>
 <test name="System.Net.Sockets.Performance.Tests.Perf_Sockets.SocketTest(port: 0, bufferSize: 256, socketInstances: 1, serverType: Async, clientType: Async)" type="System.Net.Sockets.Performance.Tests.Perf_Sockets" method="SocketTest" time="0.2610967" result="Pass">
	<traits>
	  <trait name="Benchmark" value="true" />
	</traits>
	<performance runid="System.Net.Sockets.Async.Performance.Tests.dll-Linux" etl="/home/ianha/sockets/results/System.Net.Sockets.Async.Performance.Tests.dll-Linux.csv">
	  <metrics>
		<Duration displayName="Duration" unit="msec" />
	  </metrics>
	  <iterations>
		<iteration index="0" Duration="0.29530000686645508" />
		<iteration index="1" Duration="0.17589998245239258" />
		<iteration index="2" Duration="0.16830003261566162" />
		<iteration index="3" Duration="0.15550005435943604" />
		<iteration index="4" Duration="0.15030002593994141" />
		<iteration index="5" Duration="0.17829990386962891" />
		<iteration index="6" Duration="0.16260004043579102" />
		<iteration index="7" Duration="0.15530002117156982" />
		<iteration index="8" Duration="0.14859998226165771" />
		<iteration index="9" Duration="0.14769995212554932" />
		<iteration index="10" Duration="0.16190004348754883" />
		<iteration index="11" Duration="0.14670002460479736" />
		<iteration index="12" Duration="0.1864999532699585" />
		<iteration index="13" Duration="0.17809998989105225" />
		<iteration index="14" Duration="0.1576000452041626" />
		<iteration index="15" Duration="0.16360008716583252" />
		<iteration index="16" Duration="0.16120004653930664" />
		<iteration index="17" Duration="0.18369996547698975" />
		<iteration index="18" Duration="0.15650010108947754" />
		<iteration index="19" Duration="0.14800000190734863" />
		<iteration index="20" Duration="0.16309988498687744" />
		<iteration index="21" Duration="0.19889998435974121" />
		<iteration index="22" Duration="0.15230000019073486" />
		<iteration index="23" Duration="0.21770000457763672" />
		<iteration index="24" Duration="0.15880000591278076" />
		<iteration index="25" Duration="0.15750002861022949" />
		<iteration index="26" Duration="0.15309989452362061" />
		<iteration index="27" Duration="0.2146000862121582" />
		<iteration index="28" Duration="0.20319998264312744" />
		<iteration index="29" Duration="0.18610000610351563" />
		<iteration index="30" Duration="0.24790000915527344" />
		<iteration index="31" Duration="0.17590010166168213" />
		<iteration index="32" Duration="0.1521000862121582" />
		<iteration index="33" Duration="0.70919990539550781" />
		<iteration index="34" Duration="0.15429997444152832" />
		<iteration index="35" Duration="0.41680002212524414" />
		<iteration index="36" Duration="0.15810000896453857" />
		<iteration index="37" Duration="0.1622999906539917" />
		<iteration index="38" Duration="0.14529991149902344" />
		<iteration index="39" Duration="0.15430009365081787" />
		<iteration index="40" Duration="0.14859998226165771" />
		<iteration index="41" Duration="0.14600002765655518" />
		<iteration index="42" Duration="0.19299995899200439" />
		<iteration index="43" Duration="0.1520000696182251" />
		<iteration index="44" Duration="0.14490008354187012" />
		<iteration index="45" Duration="0.14289999008178711" />
		<iteration index="46" Duration="0.14030003547668457" />
		<iteration index="47" Duration="0.15779995918273926" />
		<iteration index="48" Duration="0.1491999626159668" />
		<iteration index="49" Duration="0.15810000896453857" />
		<iteration index="50" Duration="0.14840006828308105" />
	  </iterations>
	</performance>
  </test>
```

How does that correspond with your overall timing data?